apps/dashboard/test/models/announcements_test.rb

```diff
@@ -77,6 +77,17 @@ class AnnouncementsTest < ActiveSupport::TestCase
     assert_equal 0, announcements.select(&:valid?).count
   end
 
+  test "should create announcement for syntax error" do
+    f = Tempfile.open(["announcemment", ".yml"])
+    f.write "<%- end %>"
+    f.close
+
+    announcements = Announcements.all(f.path)
+    announcement = announcements.first
+    assert_equal :warning, announcement.type
+    assert_send([announcement.instance_variable_get(:@error).to_str, :include?, "syntax error"])
```

Comment:
OK - I now see this is where you were using it!

If you're going to write a test for this, then maybe you should do something like this

```ruby
Rails.logger.stubs(:warn).with("Syntax error in announcement ...").returns(0)
```
that stubs the warning log line with the specific message and the test will fail if `Rails.logger.warn` is either called with different arguments (a different string) OR is never called at all.